src/main/java/org/opentripplanner/middleware/models/OtpUser.java

```diff
@@ -143,20 +148,34 @@ public boolean delete(boolean deleteAuth0User) {
         // If a related user, invalidate relationship with all dependents.
         for (String userId : dependents) {
             OtpUser dependent = Persistence.otpUsers.getById(userId);
-            if (dependent != null) {
-                for (RelatedUser relatedUser : dependent.relatedUsers) {
-                    if (relatedUser.email.equals(this.email)) {
-                        relatedUser.status = RelatedUser.RelatedUserStatus.INVALID;
-                    }
-                }
+            if (dependent != null && invalidateRelatedUsers(email, dependent.relatedUsers)) {
                 Persistence.otpUsers.replace(dependent.id, dependent);
             }
         }
 
-        // If a dependent, remove relationship with all related users.
+        // If a dependent user, remove relationship with all related users.
         for (RelatedUser relatedUser : relatedUsers) {
             removeDependent(this, relatedUser);
         }
+
+        // If a dependent user, remove self as the primary traveler in trips created by other users.
+        // TODO: Should we alert the user who created the trip of the deletion?
+        Persistence.monitoredTrips
+            .getFiltered(Filters.eq("primary.userId", id))
+            .forEach(trip -> removePrimaryTraveler(this, trip));
+
+        // If a companion user, invalidate relationship in trips where they are companions and observers.
+        // TODO: Should we alert the user who created the trip of the deletion?
+        Persistence.monitoredTrips
+            .getFiltered(Filters.eq("companion.email", email))
+            .forEach(trip -> removeCompanion(this, trip));
+
+        // If a companion user, invalidate relationship in trips where they are companions and observers.
+        // TODO: Should we alert the user who created the trip of the deletion?
```

Comment:
I don't think it would hurt to notify the creator of the trip.

Reply:
Created issue #283 to track that. The dilemma is to choose between #282 and #283.